pkg/packagekit/updates.jsx

```diff
@@ -226,15 +251,25 @@ class ApplyUpdates extends React.Component {
     }
 
     componentDidMount() {
-        var transProxy = this.props.transaction;
-
-        transProxy.addEventListener("Package", (event, info, packageId) => {
-            var pfields = packageId.split(";");
-            // info: see PK_STATUS_* at https://github.com/hughsie/PackageKit/blob/master/lib/packagekit-glib2/pk-enum.h
-            this.setState({curPackage: pfields[0] + " " + pfields[1],
-                           curStatus: info,
-                           percentage: transProxy.Percentage <= 100 ? transProxy.Percentage : 0,
-                           timeRemaining: transProxy.RemainingTime > 0 ? transProxy.RemainingTime : null});
+        var transObj = this.props.transaction;
+
+        pkWatchTransaction(transObj, {
+            Package: (info, packageId) => {
+                var pfields = packageId.split(";");
+
+                dbus_pk.call(transObj, "org.freedesktop.DBus.Properties", "GetAll", [I_TRANS], {timeout: 500})
```

Comment:
For starting transactions a long timeout (5s) is fine. But this is a handler for PackageKit's "progress" signals, which occur very often (magnitude of once per second). These handlers shouldn't overlap (too much), this is why I set a low timeout here. It's better to miss one call (just in case) than eternally limping behind. I added a comment to explain.